templates/tower-project.yaml

```diff
@@ -0,0 +1,338 @@
+
+  TowerBucketPolicy:
+    Type: AWS::S3::BucketPolicy
+    Properties:
+      Bucket: !Ref TowerBucket
+      PolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Sid: DenyAccessToNonAdmins
+            Effect: Deny
+            Principal:
+              AWS: !Ref NonAdminArns
+            Action: "s3:*"
+            Resource:
+              - !Sub "arn:aws:s3:::${TowerBucket}"
+              - !Sub "arn:aws:s3:::${TowerBucket}/*"
+          - !If
+            - HasReadWriteAccessArns
+            - Sid: GrantWriteAccessToReadWriteAccessArns
+              Effect: Allow
+              Principal:
+                AWS: !Ref ReadWriteAccessArns
+              Action:
+                - "s3:PutObject*"
+              Condition:
+                StringEquals:
+                  s3:x-amz-acl: bucket-owner-full-control
+              Resource:
+                - !Sub "arn:aws:s3:::${TowerBucket}/*"
+            - !Ref AWS::NoValue
+          - !If
+            - HasReadWriteAccessArns
+            - Sid: GrantRemainingAccessToReadWriteAccessArns
+              Effect: Allow
+              Principal:
+                AWS: !Ref ReadWriteAccessArns
+              Action:
+                - "s3:GetObject*"
+                - "s3:DeleteObject*"
+                - "s3:*MultipartUpload*"
+                - "s3:ListBucket*"
+                - "s3:GetBucketLocation"
+              Resource:
+                - !Sub "arn:aws:s3:::${TowerBucket}"
+                - !Sub "arn:aws:s3:::${TowerBucket}/*"
+            - !Ref AWS::NoValue
+          - !If
+            - HasReadOnlyAccessArns
+            - Sid: GrantReadAccessToReadOnlyAccessArns
+              Effect: Allow
+              Principal:
+                AWS: !Ref ReadOnlyAccessArns
+              Action:
+                - "s3:ListBucket*"
+                - "s3:GetBucketLocation"
+                - "s3:GetObject"
+                - "s3:GetObjectAcl"
+              Resource:
+                - !Sub "arn:aws:s3:::${TowerBucket}"
+                - !Sub "arn:aws:s3:::${TowerBucket}/*"
+            - !Ref AWS::NoValue
+
+  TowerForgeServiceUser:
+    Type: AWS::IAM::User
+    Properties:
+      ManagedPolicyArns:
+        - !Ref TowerForgePolicyArn
+        - !Ref TowerLaunchPolicyArn
+      Policies:
+        - PolicyName: TowerForgeServiceUserS3Access
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              Effect: Allow
+              Action:
+                - "s3:ListBucket"
+                - "s3:GetObject"
+                - "s3:PutObject"
+                - "s3:PutObjectTagging"
+                - "s3:DeleteObject"
+              Resource:
+                - !Sub "arn:aws:s3:::${TowerBucket}"
+                - !Sub "arn:aws:s3:::${TowerBucket}/*"
+
+  TowerForgeServiceUserAccessKey:
+    Type: AWS::IAM::AccessKey
+    Properties:
+      UserName: !Ref TowerForgeServiceUser
+
+  TowerProjectConfigurationSecret:
```

Comment:
I'm now storing various configuration details in this secret since users will no longer have access to CloudFormation output values (due to the fact that I can't bind a resource-based policy to a CFN stack).